src/main/java/graphql/execution/incremental/DeferredCall.java

```diff
@@ -0,0 +1,135 @@
+package graphql.execution.incremental;
+
+import com.google.common.collect.ImmutableList;
+import graphql.ExecutionResult;
+import graphql.GraphQLError;
+import graphql.Internal;
+import graphql.execution.Async;
+import graphql.execution.NonNullableFieldWasNullError;
+import graphql.execution.NonNullableFieldWasNullException;
+import graphql.execution.ResultPath;
+import graphql.incremental.DeferPayload;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.function.Supplier;
+
+/**
+ * Represents a deferred call (aka @defer) to get an execution result sometime after the initial query has returned.
+ * <p>
+ * A deferred call can encompass multiple fields. The deferred call will resolve once all sub-fields resolve.
+ * <p>
+ * For example, this query:
+ * <pre>
+ * {
+ *     post {
+ *         ... @defer(label: "defer-post") {
+ *             text
+ *             summary
+ *         }
+ *     }
+ * }
+ * </pre>
+ * Will result on 1 instance of `DeferredCall`, containing calls for the 2 fields: "text" and "summary".
+ */
+@Internal
+public class DeferredCall implements IncrementalCall<DeferPayload> {
```

Comment:
Naming is hard but this (ans the ENF version) should be some other name

* DeferredBlockCall
* DeferredFragmentCall
* DeferredFragmentInvocation